codalab/rest/util.py

```diff
@@ -0,0 +1,65 @@
+"""
+Helper functions for working with the BundleModel.
+Most of these are adapted from the LocalBundleClient methods,
+Placed in this central location to prevent circular imports.
+"""
+import httplib
+import os
+import re
+
+from bottle import abort, local, request
+from marshmallow_jsonapi import Schema
+
+from codalab.common import UsageError, PermissionError, NotFoundError
+from codalab.lib import worksheet_util, spec_util, canonicalize, formatting
+from codalab.model.util import LikeQuery
+from codalab.objects.permission import unique_group
+
+
+def get_resource_ids(document, type_):
+    links = document['data']
+    if not isinstance(links, list):
+        links = [links]
+    if any(link['type'] != type_ for link in links):
+        raise UsageError('type must be %r' % type_)
+    return [link['id'] for link in links]
+
+
+#############################################################
+# GROUPS
+#############################################################
+
+
+def ensure_unused_group_name(name):
```

Comment:
Please avoid copying code as much as possible. You can change the local_bundle_client to use these methods.